src/uu/split/src/split.rs

```diff
@@ -65,6 +67,86 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     }
 }
 
+/// If the same separator (with the same value) was used multiple times - `split` should NOT fail
+/// If the separator was used multiple times but with different values - `split` should fail
+/// Unfortunately Clap cannot support both conditions simultaneously, so it is setup to fail on the second condition
+/// For the first condition - this function will find if there are multiple separator options with the same value
+/// and remove duplicates, leaving only one
+/// i.e. `split -t: -t:` will become `split -t:`
+/// However it will leave other values untouched, so
+/// `split -t: -t: -t,` will become `split -t: -t,` and Clap will panic next as it should
+fn handle_multiple_separator_options(args: &[OsString]) -> Vec<OsString> {
+    let mut separator_values: Vec<&str> = vec![];
+    let mut preceding_sep_opt_req_value = false;
+    let mut i = -1;
+    let mut slices_to_remove: Vec<i32> = vec![];
+
+    // collect slices to remove
+    for os_slice in args {
```

Comment:
you probably can do the i this way:

    for (i, os_slice) in args.iter().enumerate() {

Reply:
possibly, yes. I actually switched to the current form based on clippy suggestion. Will see if this can be refactored further